cmd/kubeadm/app/util/version.go

```diff
@@ -40,7 +40,7 @@ const (
 var (
 	kubeReleaseBucketURL  = "https://dl.k8s.io"
 	kubeReleaseRegex      = regexp.MustCompile(`^v?(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)([-0-9a-zA-Z_\.+]*)?$`)
-	kubeReleaseLabelRegex = regexp.MustCompile(`^[[:lower:]]+(-[-\w_\.]+)?$`)
+	kubeReleaseLabelRegex = regexp.MustCompile(`^[a-z0-9]+(-[-\w_\.]+)?$`)
```

Comment:
both the old and new regexp are not optimal.

please use this:
https://play.golang.org/p/zprnyzsKDnV

and add a unit test for `k8s-master`.